From 19f3a213b13274e107c2ed0e5d7436aef27ade4e Mon Sep 17 00:00:00 2001
From: Oliver Neukum <oneukum@suse.com>
Date: Thu, 3 Sep 2015 13:37:22 +0200
Subject: [PATCH] ehci-pci: enable interrupt on BayTrail
References: bnc#947337
Patch-Mainline: Never (rendered obsolete by later changes in the PCI layer)

BayTrail needs the interrupt to be explicitely enabled
in our old 3.0 framework

Signed-off-by: Oliver Neukum <oneukum@suse.com>
---
 drivers/usb/host/ehci-pci.c |    8 ++++++++
 1 file changed, 8 insertions(+)

--- a/drivers/usb/host/ehci-pci.c
+++ b/drivers/usb/host/ehci-pci.c
@@ -133,6 +133,14 @@ static int ehci_pci_setup(struct usb_hcd
 				ehci->no_selective_suspend = 1;
 			break;
 		}
+		/*
+		 * on some BayTrail systems the firmware hands over the
+		 * EHCI with interrupts disabled. Just in case, we override.
+		 * Failure to do so renders the affected controllers unusable.
+		 * In later kernels the generic PCI layer does this.
+		 */
+		if (pdev->device == 0x0f34)
+			pci_intx(pdev, 1);
 		break;
 	case PCI_VENDOR_ID_INTEL:
 		if (pdev->device == PCI_DEVICE_ID_INTEL_CE4100_USB)
